driver/main.c

```diff
@@ -2261,32 +2300,51 @@ TRACEPOINT_PROBE(syscall_exit_probe, struct pt_regs *regs, long ret)
 	 * which is a very old syscall, not used anymore by most applications
 	 */
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 9, 0)
-	if (in_ia32_syscall() && id != __NR_execve) {
+	if (in_ia32_syscall() && event_data.event_info.syscall_data.id != __NR_execve) {
 #else
-	if (unlikely((task_thread_info(current)->status & TS_COMPAT) && id != __NR_execve)) {
+	if (unlikely((task_thread_info(current)->status & TS_COMPAT) && event_data.event_info.syscall_data.id != __NR_execve)) {
 #endif
 		event_data.event_info.syscall_data.cur_g_syscall_table = g_syscall_ia32_table;
 		event_data.compat = true;
-		socketcall_syscall = __NR_ia32_socketcall;
+		event_data.socketcall_syscall_id = __NR_ia32_socketcall;
 	}
 #endif
 
 	g_n_tracepoint_hit_inc();
 
 #ifdef _HAS_SOCKETCALL
-	if (id == socketcall_syscall)
+	if(event_data.event_info.syscall_data.id == event_data.socketcall_syscall_id)
 	{
-		id = convert_network_syscalls(regs, socketcall_syscall);
-		if (id != socketcall_syscall)
+		bool is_syscall_return;
+		int return_code = convert_network_syscalls(regs, &is_syscall_return);
+
+		/* If the return code is not the generic event we will need to extract parameters
+		 * with the socket call mechanism.
+		 */
+		event_data.extract_socketcall_params = true;
+
+		/* If we return an event code, it means we need to call directly `record_event_all_consumers` */
+		if(!is_syscall_return)
 		{
-			event_data.is_socketcall = true;
+			/* The user provided a wrong code, we will send a generic event, 
+			 * no need for socket call arguments extraction logic.
+			 */
+			if(return_code == PPME_GENERIC_E)
+			{
+				event_data.extract_socketcall_params = false;
+			}
+			/* we need to use `return_code + 1` because return_code
+			 * is the enter event.
+			 */
+			record_event_all_consumers(return_code + 1, return_code == PPME_GENERIC_E ? UF_ALWAYS_DROP : UF_USED, &event_data, KMOD_PROG_SYS_EXIT);
```

Comment:
Missing `return` statement?

Reply:
Btw i'd avoid this at all, just setting data to be used by the `record_event_all_consumers` at the end of the function. Same for sys enter.

Reply:
Great catch!